src/math/trig.c

```diff
@@ -94,9 +122,212 @@ void cordic_sin_cos(int32_t th_rad_fxp, cordic_cfg type, int32_t *sign, int32_t
+ * Description	: Determines input value is scalar [logical 1 (true) logical 0 (false)]
+ */
+static int32_t iscalarcordicasin(int32_t sinvalue, int16_t numiters,
+				 const int32_t LUT[CORDIC_30B_ITABLE_SIZE])
+{
+	int32_t b_i;
+	int32_t b_sinvalue;
+	int32_t i;
+	int32_t x;
+	int32_t xdshift;
+	int32_t xshift;
+	int32_t y;
+	int32_t ydshift;
+	int32_t yshift;
+	int32_t z;
+	int16_t j;
+	int16_t k;
+	b_sinvalue = (sinvalue >> 1);
+	if (b_sinvalue > cord_arcsincos_q28fl) {
+		x = 0;
+		y = cord_arcsincos_q30fl;
+		z = cord_arcsincos_piovertwo_q29fl;
+	} else {
+		x = cord_arcsincos_q30fl;
+		y = 0;
+		z = 0;
+	}
+	b_sinvalue <<= 1;
+	i = (int32_t)((int16_t)(numiters - 1));
+	for (b_i = 0; b_i < i; b_i++) {
+		j = (int16_t)((b_i + 1) << 1);
+		if (j >= 31) {
+			j = 31;
+		}
+		if (b_i < 31) {
+			k = (int16_t)b_i;
+		} else {
+			k = 31;
+		}
+		xshift = (x >> ((uint32_t)k));
+		xdshift = (x >> ((uint32_t)j));
+		yshift = (y >> ((uint32_t)k));
+		ydshift = (y >> ((uint32_t)j));
+		if (y == b_sinvalue) {
+			x += xdshift;
+			y += ydshift;
+		} else if (((y >= b_sinvalue) && (x >= 0)) ||
+			   ((y < b_sinvalue) && (x < 0))) {
+			x = (x - xdshift) + yshift;
+			y = (y - ydshift) - xshift;
+			z -= LUT[b_i];
+		} else {
+			x = (x - xdshift) - yshift;
+			y = (y - ydshift) + xshift;
+			z += LUT[b_i];
+		}
+		b_sinvalue += (b_sinvalue >> ((uint32_t)j));
+	}
+	if (z < 0) {
+		z = -z;
+	}
+	return z;
+}
+ /**
+  * \CORDIC-based approximation of inverse sine
+  * \Arguments	: int32_t cdc_acos_th inverse sine of cdc_asine_th based on a
+  * \		  CORDIC approximation containing real values from -1 to 1 in Q2.30
+  * \Return Type: uint32_t
+  * \DEscription: Regulated range is between [3.14159265346825 to 0] in radians Q3.29
+  *               'Error (max = 0.000000013422120), THD+N  = -157.443577601060156'
+  */
+uint32_t asin_fixed_32b(int32_t cdc_asin_th)
+{
+	int32_t saturatedunaryminus;
+	if (cdc_asin_th >= 0) {
+		saturatedunaryminus =
+		    iscalarcordicasin(cdc_asin_th, (int16_t)31, cordic_ilookup);
+	} else {
+		if (cdc_asin_th <= INT32_MIN) {
+			saturatedunaryminus = INT32_MAX;
+		} else {
+			saturatedunaryminus = -cdc_asin_th;
+		}
+		saturatedunaryminus =
+		    -iscalarcordicasin(saturatedunaryminus, (int16_t)31, cordic_ilookup);
+	}
+	if (saturatedunaryminus < 0) {
+		saturatedunaryminus = 0;
+	}
+	return (uint32_t)saturatedunaryminus;
+}
+/**
+ * Arguments	: int32_t cosvalue Q2.30
+ *		  int16_t numiters
+ *		  const int32_T LUT[30]
+ * Return Type	: int32_t
+ * Description	: Determines input value is scalar [logical 1 (true) logical 0 (false)]
+ */
+static int32_t iscalarcordicacos(int32_t cosvalue, int16_t numiters,
+				 const int32_t LUT[CORDIC_30B_ITABLE_SIZE])
+{
+	int32_t b_cosvalue;
+	int32_t b_i;
+	int32_t i;
+	int32_t x;
+	int32_t xdshift;
+	int32_t xshift;
+	int32_t y;
+	int32_t ydshift;
+	int32_t yshift;
+	int32_t z;
+	int16_t j;
+	int16_t k;
+	b_cosvalue = (cosvalue >> 1);
+	if (b_cosvalue < cord_arcsincos_q28fl) {
```

Comment:
hmm, this seems duplicated from the function above, i wonder if we can optimize this